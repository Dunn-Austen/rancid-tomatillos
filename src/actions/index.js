export const addMovies = movies => ({
  type: 'ADD_MOVIES',
  movies: movies
});

export const updateUser = user => ({
  type: 'UPDATE_USER',
  user: user
});

export const updateLoggedInStatus = status => ({
  type: 'UPDATE_LOGGEDIN_STATUS',
  status: status
});

export const handleError = errorMessage => ({
  type: 'HANDLE_ERROR',
  errorMessage
});

export const isLoading = loadingStatus => ({
  //Passing in a boolean, where loadingStatus key is assigned: true/false
  type: 'IS_LOADING',
  loadingStatus
});
