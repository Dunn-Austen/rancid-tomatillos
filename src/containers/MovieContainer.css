.movie-container {
  background: linear-gradient(rgba(0, 0, .5, .5), rgba(0, .1, .1, .1)), url("../images/curtains.jpeg");
  background-blend-mode: lighten saturation;
  background-repeat: no-repeat;
  background-size: 100% 100%;
  display: flex;
  height: 92vh;
  justify-content: center;
  width: 100%;
}

.inner-container {
  background-color: #3B397F;
  border: 10px ridge #c654dd;
  box-shadow: 0 0 20px 10px black;
  display: flex;
  flex-wrap: wrap;
  height: 90%;
  justify-content: center;
  margin-top: 20px;
  overflow: scroll;
  width: 78%;
}

.sort-button {
  background-color: #2D2C2F;
  border: 0.1px groove #c554dd;
  border-radius: 10px;
  box-shadow: 0 0 1px 1px white;
  color: #ED7899;
  font-size: 20px;
  font-weight: bold;
  height: 30px;
  line-height: 5px;
  margin-top: 10px;
  outline: none;
  width: 295px;
}

.sort-button:hover {
  background-color: #c654dd;
  border: 1px groove white;
  color: black;
}
