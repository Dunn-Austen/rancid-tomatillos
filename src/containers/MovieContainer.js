import React from 'react';
import { connect } from 'react-redux';

const MovieContainer = ({ allMovies }) => {
  console.log('from MovieContainer', allMovies);
  return (
    <div>
      <h1>MovieContainer</h1>
    </div>
  );
};

export default connect(null, null)(MovieContainer);
