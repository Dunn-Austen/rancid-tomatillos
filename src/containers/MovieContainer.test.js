import React from 'react';
import { shallow } from 'enzyme';
import { MovieContainer, mapStateToProps } from './MovieContainer';
import { addMovies } from '../actions'

describe('MovieContainer', () => {
  const mockAddMovies = jest.fn();
  const mockMoviesData = [
    {
      id: 1,
      title: 'Jumanji: The Next Level',
      poster_path:
        'https://image.tmdb.org/t/p/original//l4iknLOenijaB85Zyb5SxH1gGz8.jpg',
      backdrop_path:
        'https://image.tmdb.org/t/p/original//zTxHf9iIOCqRbxvl8W5QYKrsMLq.jpg',
      release_date: '2019-12-04',
      overview:
        "In Jumanji: The Next Level, the gang is back but the game has changed. As they return to rescue one of their own, the players will have to brave parts unknown from arid deserts to snowy mountains, to escape the world's most dangerous game.",
      average_rating: 4
    },
    {
      id: 2,
      title: 'Ad Astra',
      poster_path:
        'https://image.tmdb.org/t/p/original//xBHvZcjRiWyobQ9kxBhO6B2dtRI.jpg',
      backdrop_path:
        'https://image.tmdb.org/t/p/original//5BwqwxMEjeFtdknRV792Svo0K1v.jpg',
      release_date: '2019-09-17',
      overview:
        'The near future, a time when both hope and hardships drive humanity to look to the stars and beyond. While a mysterious phenomenon menaces to destroy life on planet Earth, astronaut Roy McBride undertakes a mission across the immensity of space and its many perils to uncover the truth about a lost expedition that decades before boldly faced emptiness and silence in search of the unknown.',
      average_rating: 7
    },
    {
      id: 3,
      title: 'Frozen II',
      poster_path:
        'https://image.tmdb.org/t/p/original//pjeMs3yqRmFL3giJy4PMXWZTTPa.jpg',
      backdrop_path:
        'https://image.tmdb.org/t/p/original//xJWPZIYOEFIjZpBL7SVBGnzRYXp.jpg',
      release_date: '2019-11-20',
      overview:
        'Elsa, Anna, Kristoff and Olaf head far into the forest to learn the truth about an ancient mystery of their kingdom.',
      average_rating: 8
    }
  ];

  it('should match the snapshot with loadingStatus of false', () => {
    const wrapper = shallow(<MovieContainer
      allMovies={mockMoviesData}
      errorMessage={''}
      loadingStatus={false}
      addMovies={mockAddMovies}
    />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot with loadingStatus of true', () => {
    const wrapper = shallow(<MovieContainer
      allMovies={mockMoviesData}
      errorMessage={''}
      loadingStatus={true}
      addMovies={mockAddMovies}
    />);

    expect(wrapper).toMatchSnapshot();
  });

//Failing Test
  // it('should invoke addMovies on click',
  //   () => {
  //   const wrapper = shallow(<MovieContainer
  //     allMovies={mockMoviesData}
  //     errorMessage={''}
  //     loadingStatus={true}
  //     addMovies={mockAddMovies}
  //   />);
  //
  //   wrapper.find('.sort-button').simulate('click');
  //   expect(mockAddMovies).toHaveBeenCalled();
  // });

  describe('mapStateToProps', () => {
    it('should return an object with error, loading, and allMovies \
      properties from state', () => {
        const mockState = {
          movies: [{sample: 'datum'}],
          user: {},
          isLoggedIn: true,
          errorMessage: 'Made Mistake',
          loadingStatus: false,
        };
        const expected = {
          allMovies: [{sample: 'datum'}],
          errorMessage: 'Made Mistake',
          loadingStatus: false
        };
        const mappedProps = mapStateToProps(mockState);

        expect(mappedProps).toEqual(expected);
    });
  });

  describe('mapDispatchToProps', () => {
    it('calls dispatch with addMovies action when addMovies is called',
      () => {
        const mockDispatch = jest.fn();
        const actionToDispatch = addMovies([{}, {}]);
        const mappedProps = mapDispatchToProps(mockDispatch);

        mappedProps.addMovies([{}, {}]);

        expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  });
});
