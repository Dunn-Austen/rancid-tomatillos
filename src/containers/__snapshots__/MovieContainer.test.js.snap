// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieContainer should match the snapshot with all data successfully passed 1`] = `
<div
  className="movie-container"
/>
`;
